### PR TITLE
Mobileapps 1000

### PR DESCRIPTION
**Pull request by ankitgoyalgloballogic**

### CODE REVIEW

File: ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift [hunk 1]
```diff
@@ -163,13 +163,14 @@ extension SearchViewModel {
             // load data from bundle
             loadConfigurationsFromAppBundle()
         }
-        self.loadConfigurationFromServer()
+        // self.loadConfigurationFromServer()
```
Comment (gl-lovekesh) [inline ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift:166]: Add comment, why do you need this commented code?

File: ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift [hunk 2]
```diff
@@ -379,6 +380,14 @@ extension SearchViewModel {
     }
     
     // MARK: Updated Facet Field Options
+    func isFacetChipsHasSelectedValue() -> Bool {
+        
+        if self.searchModel.searchChips.enumerated().first(where: {!($0.element.selectedValue).isEmpty && ($0.element.componentType == .facetField || $0.element.componentType == .facetQuery || $0.element.componentType == .facetInterval)}) != nil {
```
Comment (gl-lovekesh) [inline ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift:385]: If this is an array then use higher order function directly as first.

File: ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift [hunk 3]
```diff
@@ -455,8 +471,11 @@ extension SearchViewModel {
             // Step 2: Get index of object from old intervals which matches object from new interval
             if let object = oldFacetIntervals.enumerated().first(where: {$0.element.label == newLabel}) {
```
Comment (gl-lovekesh) [inline ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift:472]: Check here as well.

File: ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift [hunk 4]
```diff
@@ -400,8 +409,12 @@ extension SearchViewModel {
             // Step 2: Get index of object from old fields which matches object from new fields
             if let object = oldFacetIFields.enumerated().first(where: {$0.element.label == newLabel}) {
```
Comment (gl-lovekesh) [inline ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift:410]: Check here for higher order function.

**Review by gl-lovekesh — CHANGES_REQUESTED** [review]

File: ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift (hunk 1 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift:166]: Done

File: ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift (hunk 2 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift:385]: Done

File: ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift (hunk 3 above)
Comment (ankitgoyalgloballogic) [reply ContentApp/BusinessLayer/Search/ViewModels/SearchViewModel.swift:472]: We need offset value for the object. So, need enumerated() here

**Review by gl-lovekesh — APPROVED** [review]

**Pull request merged**